common/src/main/java/org/keycloak/common/util/Environment.java

```diff
@@ -26,6 +26,21 @@ public class Environment {
 
     public static final int DEFAULT_JBOSS_AS_STARTUP_TIMEOUT = 300;
 
+    /*
+     * @return major version of java runtime like 8, 11 or 17
+     */
+    public static int getJavaVersion() {
```

Comment:
when this is targeted to the time we won't have support for wildfly/java8 anymore, I suggest to use `Runtime.version().major()` instead. way less verbose and gives what you want in one line of code. Available since java9.

Reply:
I've changed the quarkus part per the suggestion of @pedroigor so it doesn't use this anymore. Hence I removed the stuff from the `Environment` class of keycloak-common module and moved it to the Wildfly module.

However Wildfly module still needs to use that instead of `Runtime.version().major()` as it still needs to use Java 8. This stuff is target for the RHSSO 7.6.X and hence Wildfly and Java8 support still needs to be kept. But since this is in the Wildfly module, it will be auto-removed once we are able to remove Wildfly distribution entirely.